components/raftstore/src/store/worker/split_check.rs

```diff
@@ -270,9 +385,22 @@ where
         region: &Region,
         start_key: &[u8],
         end_key: &[u8],
+        bucket_ranges: Option<Vec<BucketRange>>,
     ) -> Result<Vec<Vec<u8>>> {
         let timer = CHECK_SPILT_HISTOGRAM.start_coarse_timer();
-        let mut bucket_keys = Vec::new();
+        let mut buckets = Vec::new();
+        let mut bucket = Bucket::default();
+        let empty_bucket = vec![];
+        let (mut skip_check_bucket, bucket_range_list) =
+            if let Some(ref bucket_range_list) = bucket_ranges {
+                (
+                    bucket_range_list.is_empty() || !host.enable_region_bucket(),
+                    bucket_range_list,
+                )
+            } else {
+                (!host.enable_region_bucket(), &empty_bucket)
+            };
```

Comment:
It looks strange. Why it skips checking bucket if `bucket_ranges` is `Some(vec![])` and may checks bucket if `bucket_ranges` is `None`.

Reply:
When specifying bucket_ranges with value, it means check_split_and_bucket will ONLY check bucket in bucket_ranges and thus it's very straightforward that if it's empty then nothing to check. 
On the other hand when bucket_ranges is not specified, it means checks the whole region. The reason to have None instead of a vec![Bucket{start, end key of region}] is that it's more explicit to mean the whole region.